cpp/src/Slice/Parser.cpp

```diff
@@ -1023,16 +1036,17 @@ Slice::Contained::isDeprecated() const
 optional<string>
 Slice::Contained::getDeprecationReason() const
 {
+    string reasonMessage;
```

Comment:
This is unrelated to the main goal of this PR, but is small, and fixes #3039.

If no message was supplied to `deprecated` metadata, this should return `nullopt`, not `""`.
And now it does!